source/simulator/stokes_matrix_free.cc

```diff
@@ -2208,6 +2216,10 @@ namespace aspect
             &&
             sim.geometry_model->has_curved_elements())
           {
+            // TODO: add in 2/3*div term, should be simple
```

Comment:
Currently I'm not including the `2/3*div` term when computing the A-block diagonal for non-box meshes with tangential boundaries. This can be easily fixed as part of this pull request. @gassmoeller or @tjhei, do you know of an example of a simple compressible computation on sphere/shell mesh with tangential boundary conditions I can use to test changes?

Reply:
we can do this in a separate PR, I guess.

Reply:
You should be able to use `tests/steinberger_viscosity.prm` that one is an octant of a 3D spherical with some tangential velocity boundary conditions. It does not have an analytical solution though (but I guess you are only interested in the iteration count?).

Reply:
Yes, that example is exactly what I need. I've added the fix to the `build_preconditioner()` function and a test (steinberger_compressible_gmg).